### PR TITLE
feat: Introducing option to authenticate with (personal access) token…

### PR DESCRIPTION
**Pull request by timdeluxe**
…, instead of username/password

### CODE REVIEW

File: lib/api/Configuration.ts [hunk 1]
```diff
@@ -134,11 +140,20 @@ export class Configuration {
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       let configurationDocument: any
       try {
-        configurationDocument = await got(configurationUrl, {
-          username: this.confluenceUser,
-          password: this.confluencePassword,
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        }).json<any>()
+        if (this.confluencePersonalAccessToken !== '') {
+          configurationDocument = await got(configurationUrl, {
```
Comment (dploeger) [inline lib/api/Configuration.ts:150]: I should've used axios or something that can create a prepared request runner with either basic auth or token auth. Then you wouldn't have to change that much.

**Review by dploeger — APPROVED** [review]
Looks good. Thanks! 🎁

File: lib/api/Configuration.ts (hunk 1 above)
Comment (timdeluxe) [reply lib/api/Configuration.ts:150]: Hätte hätte Fahrradkette ;-)
See it as room for future improvement.

**Review by dploeger — APPROVED** [review]

**Comment by timdeluxe** [comment]
@dploeger Thanks for the approval, but what about the build failure?

**Comment by dploeger** [comment]
> @dploeger Thanks for the approval, but what about the build failure?

Ohh, whoops. Didn't see that. Hmm. Interesting. Seems to me as if your changes resulted in changes in the request sent to the mock server in the tests. The mock server holds a list of "expected" requests and returns the response for them. Now there seems to be a new kind of request that isn't in that list. If it's valid, it has to be added.

The log is not really usable in the build failure, maybe run the mocha tests yourself and see the results in your IDE. Then you can check which request doesn't exist in the mock server and see if it's a valid request based on your changes. (maybe even run the tests on main first to see if your changes actually introduced this 😄)

**Pull request merged**